planning/obstacle_stop_planner/src/node.cpp

```diff
@@ -385,16 +405,26 @@ void ObstacleStopPlannerNode::searchObstacle(
       // create one step polygon for vehicle
       createOneStepPolygon(
         p_front, p_back, one_step_move_vehicle_polygon, vehicle_info, stop_param.lateral_margin);
-      debug_ptr_->pushPolygon(
-        one_step_move_vehicle_polygon, decimate_trajectory.at(i).pose.position.z,
-        PolygonType::Vehicle);
+      if (node_param_.enable_z_axis_obstacle_filtering) {
+        debug_ptr_->pushPolyhedron(
+          one_step_move_vehicle_polygon, z_axis_min, z_axis_max, PolygonType::Vehicle);
+      } else {
+        debug_ptr_->pushPolygon(
+          one_step_move_vehicle_polygon, p_front.position.z, PolygonType::Vehicle);
+      }
 
       PointCloud::Ptr collision_pointcloud_ptr(new PointCloud);
       collision_pointcloud_ptr->header = obstacle_candidate_pointcloud_ptr->header;
 
-      const auto found_collision_points = withinPolygon(
-        one_step_move_vehicle_polygon, stop_param.stop_search_radius, prev_center_point,
-        next_center_point, slow_down_pointcloud_ptr, collision_pointcloud_ptr);
+      const auto found_collision_points =
+        node_param_.enable_z_axis_obstacle_filtering
+          ? withinPolyhedron(
+              one_step_move_vehicle_polygon, stop_param.stop_search_radius, prev_center_point,
+              next_center_point, slow_down_pointcloud_ptr, collision_pointcloud_ptr, z_axis_min,
+              z_axis_max)
+          : withinPolygon(
```

Comment:
@lchojnack @xmfcx  In my understanding, we can choose `withinPolyhedron` or `withinPolygon` by option `enable_z_axis_obstacle_filtering`. So, I think we should add `withinPolyhedron` here.

https://github.com/autowarefoundation/autoware.universe/blob/7ac34852ea37ec3084060004f505bd61698b65f0/planning/obstacle_stop_planner/src/node.cpp#L480-L483

Reply:
@satoshi-ota Thank you for your review. I've applied the hint and I've added if `enable_z_axis_obstacle_filtering` option to another part of the code.

Reply:
@lchojnack Yes, it is :+1: Thanks :smile: